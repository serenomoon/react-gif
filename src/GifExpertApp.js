import React, { useState } from 'react'
import { CategoryAdd } from './components/CategoryAdd';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    const [category, setCategory] = useState(['One Punch']);


  return (
    <>
        <h2>GifExpertApp</h2>
        <CategoryAdd setCategory={setCategory}/>
        <hr></hr>

        <ol>
            {
                category.map(category => (
                    <GifGrid 
                        key={category}
                        category={category}
                    />
                ))
            }

        </ol>

    </>
  )
};

