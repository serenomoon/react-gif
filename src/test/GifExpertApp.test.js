import { shallow } from 'enzyme';
import React from 'react';
import { GifExpertApp } from '../GifExpertApp';

describe('Test <GifExpertApp />', () => {
    
    test('should show correct snapshot', () => {
        
        const wrapper = shallow( < GifExpertApp /> );

        expect( wrapper ).toMatchSnapshot();

    });

    test('should a list of categories', () => {
        
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow( < GifExpertApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );

    });

});