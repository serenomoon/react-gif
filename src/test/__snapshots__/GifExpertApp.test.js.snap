// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifExpertApp /> should a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Test <GifExpertApp /> should show correct snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
